apis/v1alpha2/gatewayclass_types.go

```diff
@@ -0,0 +1,196 @@
+	//
+	// ParametersRef can reference a standard Kubernetes resource, i.e. ConfigMap,
+	// or an implementation-specific custom resource. The resource can be
+	// cluster-scoped or namespace-scoped.
+	//
+	// If the referent cannot be found, the GatewayClass's "InvalidParameters"
+	// status condition will be true.
+	//
+	// Support: Custom
+	//
+	// +optional
+	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
+
+	// Description helps describe a GatewayClass with more details.
+	//
+	// +kubebuilder:validation:MaxLength=64
+	// +optional
+	Description *string `json:"description,omitempty"`
+}
+
+// ParametersReference identifies an API object containing controller-specific
+// configuration resource within the cluster.
+type ParametersReference struct {
+	// Group is the group of the referent.
+	Group Group `json:"group"`
+
+	// Kind is kind of the referent.
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Scope represents if the referent is a Cluster or Namespace scoped resource.
+	// This may be set to "Cluster" or "Namespace".
+	//
+	// +kubebuilder:validation:Enum=Cluster;Namespace
+	// +kubebuilder:default=Cluster
+	// +optional
+	Scope *string `json:"scope,omitempty"`
+
+	// Namespace is the namespace of the referent.
+	// This field is required when scope is set to "Namespace" and ignored when
+	// scope is set to "Cluster".
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+}
+
+// GatewayClassConditionType is the type for status conditions on
+// Gateway resources. This type should be used with the
+// GatewayClassStatus.Conditions field.
+type GatewayClassConditionType string
+
+// GatewayClassConditionReason defines the set of reasons that explain why a
+// particular GatewayClass condition type has been raised.
+type GatewayClassConditionReason string
+
+const (
+	// This condition indicates whether the GatewayClass has been admitted by
+	// the controller requested in the `spec.controller` field.
+	//
+	// This condition defaults to False, and MUST be set by a controller when it
+	// sees a GatewayClass using its controller string. The status of this
+	// condition MUST be set to true if the controller will support provisioning
+	// Gateways using this class. Otherwise, this status MUST be set to false.
+	// If the status is set to false, the controller SHOULD set a Message and
+	// Reason as an explanation.
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Admitted"
+	//
+	// Possible reasons for this condition to be false are:
+	//
+	// * "InvalidParameters"
+	// * "Waiting"
+	//
+	// Controllers should prefer to use the values of GatewayClassConditionReason
+	// for the corresponding Reason, where appropriate.
+	GatewayClassConditionStatusAdmitted GatewayClassConditionType = "Admitted"
```

Comment:
"Admitted" implies admission control, which this is not.  The GWC was admitted.  It just was not... accepted?  valid?   acknowledged?

Reply:
Will be fixed by #839